fmriprep/workflows/bold/base.py

```diff
@@ -837,3 +864,27 @@ def _get_series_len(bold_fname):
     skip_vols = _get_vols_to_discard(img)
 
     return img.shape[3] - skip_vols
+
+
+def _create_mem_gb(bold_fname):
+    bold_size_gb = os.path.getsize(bold_fname) / (1024**3)
+    bold_tlen = nb.load(bold_fname).shape[-1]
+    mem_gb = {
+        'filesize': bold_size_gb,
+        'resampled': bold_size_gb * 4,
+        'largemem': (0.007 * max(bold_tlen, 100) + 1.5) * bold_size_gb,
+    }
+
+    return bold_tlen, mem_gb
+
+
+def _get_wf_name(bold_fname):
+    from niworkflows.nipype.utils.filemanip import split_filename
+    fname = split_filename(bold_fname)[1]
+    fname_nosub = '_'.join(fname.split("_")[1:])
+    if 'echo' in fname_nosub:
+        fname_nosub = '_'.join(fname_nosub.split("_echo-")[:1]) + "_bold"
```

Comment:
just to double check, `_echo-` has to be the key before `_bold` always?

Reply:
That's my understanding-- @chrisfilo, please correct me if I'm wrong!